devito/ops/compiler.py

```diff
@@ -30,3 +53,130 @@ def prepare_ops(self, soname, ccode, hcode):
+        self.ops_src = '%s/%s_ops.cpp' % (self.get_jit_dir(), soname)
+        self.cache_dir = self.get_codepy_dir().joinpath(soname[:7])
+
+        # Typically we end up here
+        # Make a suite of cache directories based on the soname
+        self.cache_dir.mkdir(parents=True, exist_ok=True)
+        try:
+            with open(self.ops_src, 'r') as f:
+                self.code = f.read()
+        except FileNotFoundError:
+            warning("The file %s isn't present" % self.ops_src)
+        else:
+            self.cuda_compiler(soname)
+
+
+class CUDADeviceCompiler(CompilerOPS):
+    CXX = os.environ.get('CC', 'nvcc')
+
+    def __init__(self, *args, **kwargs):
+        super(CUDADeviceCompiler, self).__init__(*args, **kwargs)
+        self.o_ext = '.o'
+        self.cflags = ['-Xcompiler="-fPIC"', '-O3', '-g', '-gencode']
+        nv_arch = os.environ.get('NV_ARCH')
+
+        if(nv_arch is None):
+            raise ValueError("select an NVIDIA device to compile in CUDA, e.g. \
+                NV_ARCH=Kepler")
+        elif(nv_arch == 'Fermi'):
+            self.cflags.append('arch=compute_20,code=sm_21')
+        elif(nv_arch == 'Kepler'):
+            self.cflags.append('arch=compute_35,code=sm_35')
+        elif(nv_arch == 'Pascal'):
+            self.cflags.append('arch=compute_60,code=sm_60')
+        elif(nv_arch == 'Volta'):
+            self.cflags.append('arch=compute_70,code=sm_70')
+        else:
+            raise ValueError("Unknown NVIDIA architecture, select: \
+                Fermi, Kepler, Pascal or Volta")
+
+        self.ldflags = []
+
+        include_dirs = '%s %s/c/include' % (self.get_jit_dir(), self._ops_install_path)
+        self.include_dirs = include_dirs.split(' ')
+
+    def __lookup_cmds__(self):
+        self.CC = 'nvcc'
+
+
+class CudaHostCompiler(CompilerOPS):
+    CC = os.environ.get('CC', 'gcc')
+    CXX = os.environ.get('CXX', 'g++')
+    MPICC = os.environ.get('MPICC', 'mpicc')
+    MPICXX = os.environ.get('MPICXX', 'mpicxx')
+
+    def __init__(self, *args, **kwargs):
+        super(CudaHostCompiler, self).__init__(*args, **kwargs)
+        self.o_ext = '.o'
+        self._cuda_install_path = os.environ.get('CUDA_INSTALL_PATH')
+        if not self._cuda_install_path:
+            raise ValueError("Couldn't find CUDA_INSTALL_PATH \
+                environment variable, please check your CUDA installation")
+
+        cflags = '-fopenmp -O3 -fPIC -DUNIX -Wall -shared -g'
+        self.cflags = os.environ.get('CFLAGS', cflags).split(' ')
+
+        self.include_dirs.append(os.path.join(self._cuda_install_path, 'include'))
+        self.include_dirs.append(os.path.join(self._ops_install_path, 'c', 'include'))
+
+        self.library_dirs.append(os.path.join(self._cuda_install_path, 'lib64'))
+        self.library_dirs.append(os.path.join(self._ops_install_path, 'c', 'lib'))
+
+        self.libraries.append('cudart')
+        self.libraries.append('ops_cuda')
+
+    def __lookup_cmds__(self):
```

Comment:
to be removed, this overwrites the environ.get

Reply:
Same as above.